lib/npm/verify-node-gyp.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+var uuid = require('node-uuid');
+var path = require('path');
+var fs = require('fs');
+var which = require('which').sync;
+
+function VerifyNodeGyp(context) {
+  if (!(this instanceof VerifyNodeGyp))
+    return new VerifyNodeGyp(context);
+  this.verifyNodeGyp = context.module.verifyNodeGyp;
+  var id = uuid.v4();
+  var scriptFilename = id + '.js';
+  this.script = path.resolve(context.path, scriptFilename);
+  this.marker = path.resolve(context.path, id + '.tmp');
+
+  var npmLocation = which('npm');
+  var npmNodeGyp = process.platform === 'win32' ?
```

Comment:
is this consistent for all versions of npm? it seems like it could be fragile?

I might want to suggest that perhaps we enforce that users have `gyp` installed locally and have this feature fail if it is not. Thoughts?